### PR TITLE
Include hacking `.md` inside `cargo doc` output

### PR DESCRIPTION
**Pull request by yvan-sraka**

### CODE REVIEW

File: consensus/pos.md [hunk 1]
```diff
@@ -52,66 +52,65 @@ impl RollCounts {
...
-	2. draw an index among the cumsum of the `cycle_states.roll_count` for cycle `N-3` among all threads.
+    1. seed the RNG with the `Sha256(concatenation of ThreadCycleState.rng_seed for all threads in increasing order)`,
+    2. draw an index among the cumsum of the `cycle_states.roll_count` for cycle `N-3` among all threads.
 
 > The complexity of a draw should be in `O(log(K))` where K is the roll ledger size.
 
 1. if the lookback cycle N-3 < 0:
-  1. use the initial (genesis) roll registry as the roll count source
-  2. `sha256^N(cfg.initial_draw_seed)` as the seed, where `sha256^N` means that we apply the sha256 hash function N times consecutively
+1. use the initial (genesis) roll registry as the roll count source
```
Comment (AureliaDolo) [inline consensus/pos.md:82]: Wrong indentation

File: consensus/pos.md [hunk 2]
```diff
@@ -52,66 +52,65 @@ impl RollCounts {
...
+1. `sha256^N(cfg.initial_draw_seed)` as the seed, where `sha256^N` means that we apply the sha256 hash function N times consecutively
```
Comment (AureliaDolo) [inline consensus/pos.md:83]: Wrong indentation

File: consensus/pos.md [hunk 3]
```diff
@@ -52,66 +52,65 @@ impl RollCounts {
...
 
-2.  if the draw is being performed for a genesis slot
-  1. force the draw outcome to match the genesis creator's address to keep draws consistent
+1. if the draw is being performed for a genesis slot
+1. force the draw outcome to match the genesis creator's address to keep draws consistent
```
Comment (AureliaDolo) [inline consensus/pos.md:86]: Wrong indentation

**Review by AureliaDolo — COMMENTED** [review]
What happened with the indentation? There are plenty of extra spaces all over

File: consensus/endorsements.md [hunk 4]
```diff
@@ -48,58 +50,60 @@ The endorsement pool is pruned if it reaches a max size and when new slots are b
...
 ## Reward computation
 
 For each block reward R (constant or fee):
 
-- `1/(1 + config.endorsement_count)` to the block creator
-- For each included endorsement:
-    - `1/(3*(1 + config.endorsement_count))` to the block creator
-    - `1/(3*(1 + config.endorsement_count))` to the endorsed block's creator
-    - `1/(3*(1 + config.endorsement_count))` to the creator of the endorsement
+-   `1/(1 + config.endorsement_count)` to the block creator
```
Comment (AureliaDolo) [inline consensus/endorsements.md:64]: Why did you add indentation here ?

File: consensus/endorsements.md [hunk 5]
```diff
@@ -48,58 +50,60 @@ The endorsement pool is pruned if it reaches a max size and when new slots are b
...
-- block 2 at slot(2,0)
-    - created by B
-    - with operation O
-    - with following endorsements
-        - 1 endorsement by A
-        - 2 endorsement by B
-        - 3 endorsement by C
-    - constant block reward of 70 coins
+
+-   addresses A, B, C in thread 0
```
Comment (AureliaDolo) [inline consensus/endorsements.md:76]: same question on indentation

File: consensus/endorsements.md [hunk 6]
```diff
@@ -48,58 +50,60 @@ The endorsement pool is pruned if it reaches a max size and when new slots are b
...
-    - `reward * nb_created_endorsements * share = 100 * 3 * 1/21 = 14.28` because C created 3 endorsements
-- the part that goes to B is :
-    - `reward * 1/(1 + endorsement_count) = 100 * 1/(1+6) = 14.28` because B created the block
-    - `+ reward * nb_created_endorsements * share = 100 *2 * 1/21 = 9.52` because B created 2 endorsements
-    - `+ reward * endorsement_count * share = 100 * 6 * 1/21 = 28.57` because B included 6 endorsements
-    - `+ rounding`
-    - `= reward - other_stakers_parts`
-    - `= 52.39`
 
+-   the part that goes to A is :
```
Comment (AureliaDolo) [inline consensus/endorsements.md:93]: same

File: consensus/pos.md [hunk 7]
```diff
@@ -52,66 +52,65 @@ impl RollCounts {
 }
 ```
 
-* `cycle_states: Vec<VecDeque<ThreadCycleState>>` indices: [thread][cycle] -> ThreadCycleState
-where the index "cycle" is relative cycle number with respect to the latest one stored. 
+-   `cycle_states: Vec<VecDeque<ThreadCycleState>>` indices: `[thread][cycle] -> ThreadCycleState`
```
Comment (AureliaDolo) [inline consensus/pos.md:55]: why indentation ?

File: consensus/pos.md (hunk 1 above)
Comment (AureliaDolo) [reply consensus/pos.md:82]: that wasn't solved !

File: docs/api.md [hunk 8]
```diff
@@ -202,7 +199,7 @@ Get information on a block given its hash.
 - Parameters:
 
 ```javascript
-String // Block id
```
Comment (AureliaDolo) [inline docs/api.md:205]: From what I recall it must be in an array

**Review by AureliaDolo — CHANGES_REQUESTED** [review]
There is a pos.md in massa_consensus and in consensus

File: consensus/endorsements.md [hunk 9]
```diff
@@ -48,58 +50,60 @@ The endorsement pool is pruned if it reaches a max size and when new slots are b
 ## Endorsement integration
 
 When creating a block consensus asks pool for endorsements specifying :
-- target_slot: the slot of the parent in the same thread of the block been produced
-- parent: that parent's BlockId
-- creators: the ordered addresses that were selected to create endorsements in for that slot
+
+-   target_slot: the slot of the parent in the same thread of the block been produced
+-   parent: that parent's BlockId
+-   creators: the ordered addresses that were selected to create endorsements in for that slot
```
Comment (adrien-zinger) [inline consensus/endorsements.md:56]: What is the justification of all this alignments diffs?

Comment (adrien-zinger) [reply consensus/endorsements.md:56]: duplications..

File: consensus/endorsements.md (hunk 4 above)
Comment (yvan-sraka) [reply consensus/endorsements.md:64]: Default .md formating

**Comment by yvan-sraka** [comment]
https://dprint.dev/ is used as default formater that's why the indentation

File: consensus/pos.md [hunk 10]
```diff
@@ -52,66 +52,64 @@ impl RollCounts {
...
 > The complexity of a draw should be in `O(log(K))` where K is the roll ledger size.
 
 1. if the lookback cycle N-3 < 0:
-  1. use the initial (genesis) roll registry as the roll count source
-  2. `sha256^N(cfg.initial_draw_seed)` as the seed, where `sha256^N` means that we apply the sha256 hash function N times consecutively
-
-2.  if the draw is being performed for a genesis slot
-  1. force the draw outcome to match the genesis creator's address to keep draws consistent
+    1. use the initial (genesis) roll registry as the roll count source
+2. `sha256^N(cfg.initial_draw_seed)` as the seed, where `sha256^N` means that we apply the sha256 hash function N times consecutively
```
Comment (AureliaDolo) [inline consensus/pos.md:83]: Wrong identation

File: consensus/pos.md [hunk 11]
```diff
@@ -121,24 +119,24 @@ If the Option is None, all addresses are taken into account.
     1. if the block's cycle is different from its parent's => empty cycle_roll_updates
 4. set `B.roll_updates = new()`
 5. if the block is the first of a new cycle N for thread Tau:
-	1. credit `roll_price * cycle_states[Tau][1 + lookback_cycles + lock_cycles].roll_updates[addr].roll_delta` for every addr for which `roll_increment == false`
-    2. deactivate all candidate rolls for all addresses for which cycle_states[Tau][1 + lookback_cycles].production_stats[address].(.0 / (.0 + .1)) > cfg.pos_miss_rate_deactivation_threshold
+    1. credit `roll_price * cycle_states[Tau][1 + lookback_cycles + lock_cycles].roll_updates[addr].roll_delta` for every addr for which `roll_increment == false`
+    2. deactivate all candidate rolls for all addresses for which `cycle_states[Tau][1 + lookback_cycles].production_stats[address].(.0 / (.0 + .1)) > cfg.pos_miss_rate_deactivation_threshold`
```
Comment (AureliaDolo) [inline consensus/pos.md:123]: wrong indentation

**Review by AureliaDolo — CHANGES_REQUESTED** [review]

**Comment by AureliaDolo** [comment]
That may me a dumb question, but why didn't you just keep the formatting as it was ?

File: consensus/pos.md [hunk 12]
```diff
@@ -121,24 +119,24 @@ If the Option is None, all addresses are taken into account.
...
-    2. deactivate all candidate rolls for all addresses for which cycle_states[Tau][1 + lookback_cycles].production_stats[address].(.0 / (.0 + .1)) > cfg.pos_miss_rate_deactivation_threshold
-6. parse the operations of the block in order:
-	1. Apply roll updates to cur_rolls. If the new roll count under/over-flows u64 => block invalid
-	2. try chain roll updates to `B.roll_updates`. If the new roll delta under/over-flows u64 => block invalid
-	3. try chain roll updates to `cycle_roll_updates`, credit the compensation roll count as ledger changes. If any error => block invalid
-		1. try chain roll updates to `cycle_roll_updates` and gather compensation counts for each address. If error => block invalid.
-		2. try chain roll changes to `block_ledger_changes` with compensation. If error => block invalid.
+    1. credit `roll_price * cycle_states[Tau][1 + lookback_cycles + lock_cycles].roll_updates[addr].roll_delta` for every addr for which `roll_increment == false`
+6. deactivate all candidate rolls for all addresses for which `cycle_states[Tau][1 + lookback_cycles].production_stats[address].(.0 / (.0 + .1)) > cfg.pos_miss_rate_deactivation_threshold`
```
Comment (AureliaDolo) [inline consensus/pos.md:123]: aarrggpaaaaifezfez I did not saw this one

**Review by AureliaDolo — CHANGES_REQUESTED** [review]

**Review by AureliaDolo — APPROVED** [review]

**Comment by AureliaDolo** [comment]
Bors merge

**Review by AureliaDolo — APPROVED** [review]

**Comment by AureliaDolo** [comment]
bors merge

**Comment by yvan-sraka** [comment]
bors merge

**Comment by yvan-sraka** [comment]
bors merge

**Comment by yvan-sraka** [comment]
bors merge

**Comment by AureliaDolo** [comment]
bors merge

**Comment by yvan-sraka** [comment]
bors merge

**Pull request merged**